Scripts/RemediationScripts/Remediate-EnableAppServicEendToEndEncryption.ps1

```diff
@@ -0,0 +1,943 @@
+
+# Pre-requisites:
+    1. You will need Owner role on subscription.
+    2. Must be connected to Azure with an authenticated account.
+
+# Steps performed by the script:
+    To remediate:
+        1. Install and validate required modules for managing App Services.
+        2. Fetch all App Services in the subscription and check their End-to-end TLS encryption configuration.
+        3. Update non-compliant App Services to enforce End-to-end TLS encryption enabled.
+    
+    To roll back:
+        1. Validate and install the modules required to run the script.
+        2. Fetch the backup of original End-to-end TLS encryption configurations.
+        3. Restore the original End-to-end TLS encryption configurations for App Services from the backup.
+
+# Step to execute script:
+    To remediate:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to remediate resource type in the Subscription. Refer `Examples`, below.
+
+    To roll back:
+        1. Download the script.
+        2. Load the script in a PowerShell session. Refer https://aka.ms/AzTS-docs/RemediationscriptExcSteps to know more about loading the script.
+        3. Execute the script to rollback on all resource type in the Subscription. Refer `Examples`, below.
+
+# Examples:
+    To remediate:
+        1. To review the App Services in a Subscription that will be remediated:
+           Enable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -DryRun
+
+        2. To enable End-to-end TLS encryption on all App Services in a Subscription:
+           Enable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck
+
+        3. To enable End-to-end TLS encryption on all App Services in a Subscription, from a previously taken snapshot:
+           Enable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\Enable-EndToEndTLSForAppServices\AppServicesWithNonCompliant.csv
+
+        To know more about parameter execute:
+            Get-Help Enable-EndToEndTLSForAppServices -Detailed
+
+    To roll back:
+        1. To configure AllAllowed on the production slot and all non-production slots of all App Services in a Subscription, from a previously taken snapshot:
+           Disable-EndToEndTLSForAppServices -SubscriptionId 00000000-xxxx-0000-xxxx-000000000000 -PerformPreReqCheck -FilePath C:\AzTS\Subscriptions\00000000-xxxx-0000-xxxx-000000000000\202109131040\DisableEndToEndTLSForAppServices\RemediatedForAppServices.csv
+       
+        To know more about the options supported by the roll back command, execute:
+        Get-Help Disable-EndToEndTLSForAppServices -Detailed        
```

Comment:
should be Enable-EndToEndTLSForAppServices

Reply:
we are executing Disable-EndToEndTLSForAppServices for roll back not for enable it